tests/test_model.py

```diff
@@ -545,6 +546,19 @@ def test_sort(self) -> None:
         expected_uris = reorder(uris, expected_order)
         self.assertListEqual(sorted_uris, expected_uris)
 
+    def test_make_bom_link_without_bom_ref(self) -> None:
+        serial_number = uuid4()
+        version = 2
+        bom_link = XsUri.make_bom_link(serial_number, version)
+        self.assertEqual(bom_link, XsUri(f'urn:cdx:{serial_number}/{version}'))
+
+    def test_make_bom_link_with_bom_ref(self) -> None:
+        serial_number = uuid4()
+        version = 2
+        bom_ref = 'componentA'
```

Comment:
Can you please provide more guidance on how you want me to handle this?

```python
    _INVALID_URI_REGEX = re.compile(r'%(?![0-9A-F]{2})|#.*#', re.IGNORECASE + re.MULTILINE)
```
Will not allow more than one `#`, and URL encoding will result in a mismatch.

Reply:
url encode reserved characters in bom_rev:
- `#` -> `%23`
- `%` -> `%25`

Reply:
If I understand correctly, something like this? or using the `from urllib.parse import quote`
```python3
        bom_ref_part = f'#{str(bom_ref).replace("%", "%25").replace("#", "%23")}' if bom_ref else ''
        uri = f'urn:cdx:{serial_number}/{version}{bom_ref_part}'
        return cls(uri)
```

Reply:
this should do the trick

Reply:
If I am making a `bom_link` to link the component(`componentA#sub-componentB`) I will get the bom_link of `#componentA%23sub-componentB%252`. Will that be fine since they are not a string-to-string match?

Not to mention the current unit test will still fail.

Reply:
this is exactly intended.

Reply:
> If I am making a `bom_link` to link the component(`componentA#sub-componentB`) I will get the bom_link of `#componentA%23sub-componentB%252`. Will that be fine since they are not a string-to-string match?



this is exactly intended.

Reply:
Done